src/NHibernate/Tuple/DynamicMapInstantiator.cs

```diff
@@ -65,7 +74,7 @@ public bool IsInstance(object obj)
 			}
 			else
 			{
-				return false;
+				return obj is IDictionary<string, object>;
 			}
```

Comment:
If defaulting to strong typed dictionary, we would have to change the logic here too:

	var that = obj as IDictionary<string, object>;
	if (that != null)
	{
		if (entityName == null)
		{
			return true;
		}
		string type = (string)that[KEY];
		return type == null || isInstanceEntityNames.Contains(type);
	}
	else
	{
		return obj is IDictionary;
	}

Reply:
This piece code here only for `EntityMode.Map` instantiator.